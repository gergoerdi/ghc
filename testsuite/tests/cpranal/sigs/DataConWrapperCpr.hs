{-# OPTIONS_GHC -O2 -fforce-recomp #-}
module DataConWrapperCpr where

data Foo = Foo !Int

-- Should have the CPR property! Hence the wrapper $WFoo must have it.
foo :: Int -> Foo
foo = Foo
