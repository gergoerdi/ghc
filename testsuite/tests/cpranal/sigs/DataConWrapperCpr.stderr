
==================== Cpr signatures ====================
DataConWrapperCpr.$tc'Foo: #c1(#, #, #c1(#c1(#), #c1(#)), #c1(#),
                               #,
                               #c4(#c1(#c1(#, #, #c1(#c1(#), #c1(#)), #c1(#), #, #c5(*)), *),
                                   #c1(#c1(#, #, #c1(#c1(#), #c1(#)), #c1(#), #, #c5(*)), *)))
DataConWrapperCpr.$tcFoo: #c1(#, #, #c1(#c1(#), #c1(#)), #c1(#), #,
                              #c5(*))
DataConWrapperCpr.$trModule: #c1(#c1(#), #c1(#))
DataConWrapperCpr.foo: #c1(*)


